### PR TITLE
[PlaylistPlayer] Resolve plugin paths to fill the item infotag

### LINKED ISSUE #16151: Videos started using JSON API do not provide subtitle info on first playback
### Describe the bug
Here is a clear and concise description of what the problem is:

If a remote app (like Kore or Yatse) is used to start a video stream from a video add-on there is a problem with subtitles not being detected correctly on the first play. The remote app will not show available subtitles to select. If the same video is started again, the subtitles are show for selection in the app.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

- Browse a video add-on via Kore/Yatse (remote app)
- Start a video that contains subtitles from the add-on (subtitles may be set via the ListItem.setSubtitles() method)
- The video starts
- Within the remote app the available subtitles are shown and one can be selected
- Those subtitles are the same as one would see from Kodi itself

## Actual Behavior

- At the initial playback attempt, the video starts, however, within the remote app, there is no indicator or selector for subtitles: the app does not receive any subtitle info for the stream.
- If I start the same video again from the remote app, it will play again, but now it will show available subtitles and a user can select them
- Within Kodi itself the subs are listed and can be selected

## Possible Fix

Check the JSON data to see if it includes the correct subtitle information?

### To Reproduce

1. Browse a video add-on (such as Youtube) and select a video that has subtitles (you might need to configure the YT add-on to use us-en subs)
2. For the first start, no subs can be selected from the app
3. Within Kodi the subtitles can be selected from the normal GUI
4. Stop the video and start it again via the remote app
5. Now the subs do show in the remote app

### Debuglog

The debuglog can be found here: https://paste.kodi.tv/tizinajive.kodi

The logfile shows playback of the same video from the Youtube add-on twice. Once without sub info send to the remote app. And the second run where the info is available.

### Screenshots 
Here are some links or screenshots to help explain the problem:

Screenshot_20190516-210006
Screenshot of Yatse after the first playback: so sub selection

Screenshot_20190516-210021
Screenshot of Yatse after the second playback: there is a subtitle selection option

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

I tested this with Kore and Yatse on my Windows Kodi Leia install. However, I have user reports for Retrospect from users with LibreElec and Kodi on Android. So this seems to be a platform independent issue.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 10 x64
 - Kodi version: Kodi Leia 18.2 (normal installer version)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by enen92**
## Description
When a `plugin://` path is played for the first time from the jsonrpc API, there is no information stored in the database for the item and it does not contain any music or video infotags (`item->IsAudio() || item->IsVideo()` is `false`). That makes Kodi interpret the item as a playlist (`g_application.PlayMedia`) and information such as the item subtitles is not available. Users using remote control applications are not able to select the item provided subtitles. Subs will only be available the next time the item is played since at that moment the item info is already in the database and is filled by `FillFileItemList`. This PR makes sure the plugin path is resolved before checking if it is audio or video, solving the issue.

## Motivation and Context
Fixes https://github.com/xbmc/xbmc/issues/16151

## How Has This Been Tested?
With a proof of concept video plugin, changing the `plugin://` play path to make sure it is always an unknown path not in the db. Played the item via Kore remote app and checked the option to select subtitles was available.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by basrieter** [comment]
I can confirm that this fixes #16151.

File: xbmc/PlayListPlayer.cpp [hunk 1]
```diff
@@ -892,6 +894,12 @@ void PLAYLIST::CPlayListPlayer::OnApplicationMessage(KODI::MESSAGING::ThreadMess
         if (list->Size() == 1 && !(*list)[0]->IsPlayList())
         {
           CFileItemPtr item = (*list)[0];
+          // if the item is a plugin we need to resolve the URL since the infotags are only
+          // available in case the item was played before (stored on the database)
+          for (int i=0; URIUtils::IsPlugin(item->GetDynPath()) && i<5; ++i)
```
Comment (ksooo) [inline xbmc/PlayListPlayer.cpp]: Where does the magic 5 come from?

And please follow our coding guidelines. blanks before and after "=" and "<", please.

Comment (enen92) [reply xbmc/PlayListPlayer.cpp]: From nowhere :) other components in kodi do the same. It just means you'll try to resolve the url recursively for a maximum of 5 times. Plugin:// paths can resolve to plugins paths so this avoids entering in some kind of deadlock. Any suggestions?

Regarding style issues, I'll do.

**Comment by enen92** [comment]
@ksooo style issues addressed. 5 magic number explained in the comment.

**Comment by basrieter** [comment]
So, can this be merged now?

**Comment by basrieter** [comment]
Should we backport this to Leia?

**Comment by ksooo** [comment]
Sure, as the PR label "backport needed" already indicates. ;-)

**Comment by basrieter** [comment]
Didn't see that label all the way at the top!

**Comment by enen92** [comment]
@basrieter can you backport?

**Comment by basrieter** [comment]
sure, I will git it a try.

**Comment by basrieter** [comment]
I created https://github.com/xbmc/xbmc/pull/16192

**Comment by bigretromike** [comment]
I have similar issue with 
`xbmc.Player().play(url, listitem)`
When url starts with `plugin://` for the first time `Player.GetItem()` returns infovideotag, but when files is played again it does not return that info anymore. 

Does this PR fix this issue ? (I initially think so, but I build kodi 20190604-75c1d6b3e7-dirty today from master and the issue is still there)

Any chance to port his to player also (or json-rpc)?

**Comment by enen92** [comment]
@bigretromike please open a separate issue with a clear description on how to reproduce the issue. Your comment seems totally unrelated to this PR.

**Comment by bigretromike** [comment]
@enen92 I did what you asked me to: https://github.com/xbmc/xbmc/issues/16245
For me it is somehow similar, but then again from two of use you are the expert here so I don't complain.

**Pull request merged**